src/main/webapp/js/components/settings/csv.jsx

```diff
@@ -0,0 +1,319 @@
+                                        type="button"
+                                        className="btn btn-success my-2"
+                                        onClick={() => relatedFields.push({
+                                          relatedField: '',
+                                          relatedFieldCsvName: '',
+                                        })}
+                                        disabled={relatedEntityElemSize === _.size(relatedFields)}
+                                      >
+                                        Add related field
+                                      </button>
+                                    </div>
+                                  )}
+                                </FieldArray>
+                              </div>
+                            );
+                          }
+                          return renderFormField({ ...elem, name: `${name}.${elem.name}` });
+                        })}
+                        <button
+                          type="button"
+                          className="btn btn-outline-danger"
+                          onClick={() => fields.remove(index)}
+                        >
+                          Delete
+                        </button>
+                      </div>
+                    ))}
+                    <button
+                      type="button"
+                      className="btn btn-outline-success mr-2"
+                      disabled={!this.state.csv.entity}
+                      onClick={() => {
+                        this.setState({ isSelectDisabled: true });
+                        fields.push({
+                          fieldConfigId: '',
+                          csvName: '',
+                          format: '',
+                          defaultValue: '',
+                        });
+                      }}
+                    >
+                      Add field config
+                    </button>
+                    <button
+                      type="submit"
+                      className="btn btn-outline-primary m-2"
+                      disabled={invalid || !this.state.csv.entity}
+                      onClick={() => { this.onSubmit(values.csvFields, this.state.csv.entity); }}
+                    >
+                      Save
+                    </button>
+                    <button
+                      type="button"
+                      className="btn btn-outline-danger m-2"
+                      disabled={!this.state.csv.entity}
+                      onClick={() => { this.deleteConfig(); }}
+                    >
+                      Delete config
+                    </button>
+                  </div>
+                )}
+              </FieldArray>
+            </form>
+          )}
+        />
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = state => ({
+  csvConfig: state.csvConfig,
+  fieldConfig: state.fieldConfig,
```

Comment:
there are selectors to get the field config by entity type (there are separate selectors to get visible/hidden field config, you can write one to get all of them)